src/main/python/main/ayab/engine/control.py

```
@@ -265,6 +271,10 @@
         self.status.total_rows = self.pat_height
         self.status.current_row = self.pat_row + 1
         self.status.line_number = line_number
+        try:
+            self.status.memo = self.memos[self.pat_row]
+        except:
```

Ruff (0.8.2)

Do not use bare `except`

(E722)